core/src/main/scala/caliban/execution/Executor.scala

```diff
@@ -101,42 +101,63 @@ object Executor {
 
     def makeQuery(step: ReducedStep[R], errors: Ref[List[CalibanError]]): ZQuery[R, Nothing, ResponseValue] = {
 
+      def eraseError(error: CalibanError): UQuery[ResponseValue] =
+        ZQuery.fromEffect(errors.update(error :: _)).as(NullValue)
+
       @tailrec
-      def wrap(query: ZQuery[R, Nothing, ResponseValue])(
+      def wrap(query: ZQuery[R, ExecutionError, ResponseValue])(
         wrappers: List[FieldWrapper[R]],
         fieldInfo: FieldInfo
-      ): ZQuery[R, Nothing, ResponseValue] =
+      ): ZQuery[R, ExecutionError, ResponseValue] =
         wrappers match {
           case Nil => query
           case wrapper :: tail =>
             wrap(
               wrapper
                 .f(query, fieldInfo)
-                .foldM(error => ZQuery.fromEffect(errors.update(error :: _)).as(NullValue), ZQuery.succeed(_))
+                .foldM(eraseError, ZQuery.succeed(_))
```

Comment:
Why do we need to remove the error here? It should be passed to all the wrappers I think?

Reply:
That's a fair point we can all the downstream wrappers to clean up errors from the upstream.